### PR TITLE
ansible provisioner scripts for cilium.

### PR DESCRIPTION
**Pull request by mchalla**
I separated them into roles so it easily allows running a single task. Still working on the cilium install scripts, will add them tomorrow.

### CODE REVIEW

**Review by aanm — APPROVED** [review]

**Comment by aanm** [comment]
LGTM, waiting for @tgraf 's feedback

File: contrib/ansible/roles/kernel/tasks/kernel_clean.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+
+- name: Delete kernel sources
+  shell: \rm -Rf /home/vagrant/net-next
```
Comment (tgraf) [inline contrib/ansible/roles/kernel/tasks/kernel_clean.yml]: This path must be made configurable somehow

File: contrib/ansible/roles/iproute2/tasks/iproute2_build.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+---
+
+- shell: getconf _NPROCESSORS_ONLN
+  register: getconf_ncpu_output
+
+- name: Build iproute2
+  command: "{{ item }} chdir=/home/vagrant/iproute2"
```
Comment (tgraf) [inline contrib/ansible/roles/iproute2/tasks/iproute2_build.yml]: This path must be made configurable

File: contrib/ansible/roles/iproute2/tasks/iproute2_clean.yml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+
+- name: cleanup iproute2 sources
+  shell: \rm -Rf /home/vagrant/iproute2
```
Comment (tgraf) [inline contrib/ansible/roles/iproute2/tasks/iproute2_clean.yml]: This path must be made configurable

File: contrib/ansible/roles/iproute2/tasks/iproute2_fetch.yml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+---
+
+- name: Fetch bpf-wip iproute2 package
+  git: repo=git://git.breakpoint.cc/dborkman/iproute2.git
+       dest=/home/vagrant/iproute2
```
Comment (tgraf) [inline contrib/ansible/roles/iproute2/tasks/iproute2_fetch.yml]: This path must be made configurable

File: contrib/ansible/roles/iproute2/tasks/iproute2_install.yml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+---
+
+- name: install iproute2
+  shell: make install chdir=/home/vagrant/iproute2
```
Comment (tgraf) [inline contrib/ansible/roles/iproute2/tasks/iproute2_install.yml]: This path must be made configurable

File: contrib/ansible/roles/kernel/tasks/kernel_build.yml [hunk 6]
```diff
@@ -0,0 +1,30 @@
+---
+
+- shell: uname -r
+  register: uname_r_output
+
+- shell: getconf _NPROCESSORS_ONLN
+  register: getconf_ncpu_output
+
+- name: Build kernel and modules
+  command: "{{ item }} chdir=/home/vagrant/net-next"
```
Comment (tgraf) [inline contrib/ansible/roles/kernel/tasks/kernel_build.yml]: This path must be made configurable

File: contrib/ansible/roles/kernel/tasks/kernel_fetch.yml [hunk 7]
```diff
@@ -0,0 +1,7 @@
+---
+
+- name: Fetch bpf-wip kernel package
+  git: repo=git://git.breakpoint.cc/dborkman/net-next.git
+       dest=/home/vagrant/net-next
```
Comment (tgraf) [inline contrib/ansible/roles/kernel/tasks/kernel_fetch.yml]: This path must be made configurable

File: contrib/ansible/roles/kernel/tasks/kernel_install.yml [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+
+- name: Delete old kernel and modules
+  command: "{{ item }}"
+  with_items:
+    - bash -c "\rm -Rf /lib/modules/*"
+    - bash -c "\rm /boot/*-custom"
+  ignore_errors: yes
+
+- name: Install kernel and modules
+  command: "{{ item }} chdir=/home/vagrant/net-next"
```
Comment (tgraf) [inline contrib/ansible/roles/kernel/tasks/kernel_install.yml]: This path must be made configurable

**Review by tgraf — CHANGES_REQUESTED** [review]
LGTM in general. Please make the destination directory configurable

File: contrib/ansible/roles/kernel/tasks/kernel_clean.yml (hunk 1 above)
Comment (mchalla) [reply contrib/ansible/roles/kernel/tasks/kernel_clean.yml]: Ok i will see how I can make /home/vagrant configurable. Everything else is relative to it.

**Comment by tgraf** [comment]
Looks good, please rebase and clash both commits.

**Review by tgraf — APPROVED** [review]

**Pull request merged**